.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  pull_request: {}
+  push:
+    branches:
+      - master
+
+jobs:
+  tfsec:
+    name: tfsec
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: aquasecurity/tfsec-action@v1.0.3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  terraform:
+    name: Terraform
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: <2.0.0
+
+      - name: Initialize Terraform
+        run: terraform init
```

Comment:
We will go provides credentials aws to permission in bucket S3 backend 

FYI @damews

Reply:
Secrets creates in `environment: dev`

Please consider change code for way use to example below.

```yaml
environment: AWS
- name: Set AWS credentials
  uses: aws-actions/configure-aws-credentials@v1
  with:
    aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
    aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
    aws-region: ${{ secrets.AWS_REGION }}
```

Reply:
> We will go provides credentials aws to permission in bucket S3 backend
> 
> FYI @damews

Não ficou muito claro o que foi proposto aqui. Entendo que as credenciais serão adicionadas, mas presumi que ainda não era o caso. Não sendo o caso, não me faz sentido usá-las.

Quanto ao ambiente `dev`, [de acordo com a documentação](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-variables), variáveis de ambiente definidas para um environment só são acessíveis por meio do contexto `vars`, não `secrets`. O contexto `secrets` é usado para secrets no geral.

Ainda nesse tópico, porém, nos convém usar o backend S3 ou até mesmo acessar infraestrutura existente em um build de CI sem código Terraform na `master`? Em algum momento futuro, espero, é claro, termos um `terraform plan` feito nas pull requests. Se isto fosse adicionado antes do código Terraform, levaria a uma falha.

O workflow linkado no seu comentário é de _deployment_ contínuo, não de integração. Deployment está fora do escopo desse PR, ou assim que imaginava.

Um último comentário: se usaremos sempre o mesmo backend Terraform, ele pode ser adicionado como um arquivo `.tf` (e.g. `backend.tf`), assim não precisamos fazer menção explícita a ele nos workflows do GitHub Actions.

Reply:
+1

A adição do backend.tf seria interessante para comparar a infraestrutura existente. Ainda mais para um _pull request_ para a master.